### ISSUE TITLE
save() prevents harvesting on stop()?

### ISSUE THREAD

**Issue by nedbat** [coveragepy/coveragepy#79]
*Originally reported by* **Zooko O'Whielacronx (Bitbucket: [zooko](https://bitbucket.org/zooko), GitHub: [zooko](https://github.com/zooko))**

----------------------------------------

Ned:

Looking at the source code of control.py in search of a different bug, it appears to me that if you call `save()` then it will set `self._harvested = True` which means that if you then accumulate more measurements and then call `stop()` that the `stop()` will observe that `self._harvested` is already `True` and will then fail to harvest the new measurements.

----------------------------------------
- Bitbucket: https://bitbucket.org/ned/coveragepy/issue/79

**Comment by nedbat** [comment]
Hmm, I changed this logic (for other reasons), but this series of calls still won't save data:  start(), ..., save(), ... stop().  Is it important that it work?  Do you have a use case for it?

**Comment by nedbat** [comment]
Here's the test I tried that demonstrated the problem to me. It fails:

```
    def test_start_save_stop(self):
        self.make_file("code1.py", """\
            code1 = 1
            """)
        self.make_file("code2.py", """\
            code2 = 1
            code2 = 2
            """)
        cov = coverage.coverage()
        cov.start()
        self.import_local_file("code1")
        cov.save()
        self.import_local_file("code2")
        cov.stop()

        _, statements, missing, _ = cov.analysis("code1.py")
        self.assertEqual(statements, [1])
        self.assertEqual(missing, [])
        _, statements, missing, _ = cov.analysis("code2.py")
        self.assertEqual(statements, [1, 2])
        self.assertEqual(missing, [])
```

**Comment by nedbat** [comment]
*Original comment by* **Loic Dachary (Bitbucket: [dachary](https://bitbucket.org/dachary), GitHub: [dachary](https://github.com/dachary))**

----------------------------------------

It reminds me of [save() and html_report() prevent further metric gathering](https://bitbucket.org/ned/coveragepy/issues/448)

**Comment by nedbat** [comment]
Fixed in [12ea6d45d8b8 (bb)](https://bitbucket.org/ned/coveragepy/commits/12ea6d45d8b8)

**Comment by nedbat** [comment]
This is in 4.4b1.
